requests-for-comments/2025-06-18-sdk-specs.md

```diff
@@ -0,0 +1,301 @@
+   name: string
+   path: string
+   properties:
+     - description: string
+       type: string
+       name: string
+```
+
+### Reference docs pages
+
+See video demo:
+
+https://github.com/user-attachments/assets/52bcbc4e-7a3d-48e0-8453-15e196b6fb0f
+
+#### SDK ref docs headers
+Contains SDK versions, link to the raw references, and a copy as markdown button.
+![](../images/2025-06-18-sdk-specs/i1.png)
+
+For us, SDK pages will be centered around the PostHog class.
+
+#### Class methods
+Each method will have a description parsed from the **doc comment**, a more details section from the `@remarks` block, examples from the `@example` blocks, table of parameters parsed from language type definition, augmented by `@parameter` blocks.
+![](../images/2025-06-18-sdk-specs/i2.png)
+
+#### Markdown descriptions
+Each will have both a description and additional collapsible notes. There will be formatted and rendered as markdown.
+![](../images/2025-06-18-sdk-specs/i3.png)
+
+#### Full interface references
+Each parameter or return type interface, where possible, include a reference page
+![](../images/2025-06-18-sdk-specs/i4.png)
+
+#### Deprecations
+Using `@deprecated` we can indicate when methods become deprecated.
+![](../images/2025-06-18-sdk-specs/i5.png)
+
+#### Support for multiple examples and generated usage example
+Using inline `@example` comments, we can indicate key use cases. In cases where this is not available, fall back to a templated usage example.
+
+#### Links to source, where possible
+Using links like `https://unpkg.com/posthog-js@1.248.1/lib/src/types.d.ts` we can link back to the npm package source we serve using
+`unpkg.com`.
+
+#### One-off examples, special objects, etc.
+Some objects like `$event.properties` or the posthog config object don't have a good definition and we currently point to the source for these definitions. We can later augment `hogRef` to document special objects like this.
+
+#### Categories
+We will display methods by category. Init, config, and identify persons categories are always at the top. The rest sorted by product and alphabetical.
+
+### Proposed workflow
+We will automate generation, but manually version pin known-good versions to prevent breaking the website because a ref is generated in a format posthog.com can't parse.
+- References are served as jsons from GitHub
+- Each time a PR is merged to main, run a workflow to re-build json specs
+    - for JS, this means using `api-extractor` from Microsoft and a parse script maintained by us to create `hogRef`
+- Use Gatsby to fetch and build pages from version pinned jsons from GitHub
+- We will manually bump sdk version pins (or list of versions) manually
+
+## Effect on existing docs
+Where possible, for each SDK doc, hand-write usage guides for set up and key use cases only. This replaces the potentially giant pages currently used to document SDKs.
+
+## Ownership
+If you update the SDK, you're expected to update the doc comment. 
```

Comment:
Very reasonable. I'd love to see in this RFC, however, how you plan on delegating ownership in actually writing/reviewing the initial set of docs. I wouldn't expect the Docs team to be able to document all of the public interfaces we have for `posthog-js` - in the same way I wouldn't expect myself to know what each method does even though I've contributed 4k lines to it 😆

Reply:
Ah, I now see you have an outstanding PR, but that doesnt cover everything and you're simply "asking for help". I believe it makes sense for you/someone to delegate this task to people explicitly or else you'll have a hard time getting people to do this because it's time-consuming to do it "on the side" but it's possible to do it if we consider it first-class work.